File: src/main/java/org/kane/domain/service/diary/diary_record/DiaryRecordService.java
package org.kane.domain.service.diary.diary_record;

import org.kane.domain.DTO.entityDTO.diary.daily_record.DiaryRecordShowDTO;
import org.kane.domain.DTO.entityDTO.diary.sport_activity.CalorieConsumptionShowDTO;
import org.kane.domain.DTO.request.DiaryRecordRequest;

import java.security.Principal;
import java.time.LocalDate;

public interface DiaryRecordService {
    void createDiaryRecord(Principal principal,  DiaryRecordRequest diaryRecordRequest);

    DiaryRecordShowDTO showDiaryRecord(Principal principal, DiaryRecordRequest diaryRecordRequest);

    CalorieConsumptionShowDTO getConsumptionOfDiaryRecord(Principal principal, LocalDate date);
}

File: src/app/service/diary/diary-record.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import DiaryRecordRequest from '../../DTO/requests/diary-record.request';
import DiaryRecordShowDTO from '../../DTO/entity_dto/diary/diary_record/diary-record-show.dto';
import CalorieConsumptionShowDTO from '../../DTO/entity_dto/diary/sport_activity/calorie-consumption-show.dto';
import type { CalendarDate } from '../../DTO/types';
import { formatLocalCalendarDate } from '../../util/calendar-date';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class DiaryRecordService {
  private readonly http = inject(HttpClient);
  private readonly diaryAPI = new Endpoint('diary');

  createDiaryRecord(request: DiaryRecordRequest):Observable<any>{
    return this.http.put(
      this.diaryAPI.builder()
        .points("create")
        .build(),
      request)
  }

  showDiaryRecord(request: DiaryRecordRequest):Observable<DiaryRecordShowDTO>{
    return this.http.post<DiaryRecordShowDTO>(
      this.diaryAPI.builder()
        .points("show")
        .build(),
      request)
  }

  getConsumptionOfDiaryRecord(
    date: CalendarDate | Date,
  ): Observable<CalorieConsumptionShowDTO> {
    const dateParam =
      typeof date === 'string' ? date : formatLocalCalendarDate(date);
    return this.http.get<CalorieConsumptionShowDTO>(
      this.diaryAPI.builder()
        .points("consumption")
        .addParam("date", encodeURIComponent(dateParam))
        .build(),
    );
  }
}

